ruby/vampires.rb

```diff
@@ -0,0 +1,97 @@
+# Methods to check user input
+def check_if_integer
+  begin
+    num = Integer(gets.chomp)  
+  rescue 
+    puts 'Error reading input - please enter an integer number:'
+    retry
+  end
+
+  num
+end
+
+def yes_or_no
+  loop do
+    case gets.chomp.downcase
```

Comment:
Right!  Generally we pass parameters through our driver code which can be considered a stylistic decision for the time being.  It will become more apparent when you start working with web application which separate your files into models, views, and controllers (MVC).

If you pass in a parameter then you can also extract your loop into your driver code which will give a smoother design flow and solve that problem.

That said, Im glad to see that you have a good understanding of explicit return usage!